cmd/object-api-interface.go

```diff
@@ -58,4 +59,9 @@ type ObjectLayer interface {
 	ListObjectsHeal(bucket, prefix, marker, delimiter string, maxKeys int) (ListObjectsInfo, error)
 	ListUploadsHeal(bucket, prefix, marker, uploadIDMarker,
 		delimiter string, maxUploads int) (ListMultipartsInfo, error)
+
+	// Locking operations
+	ListLocks(bucket, prefix string, duration time.Duration) ([]VolumeLockInfo, error)
+	ClearLocks([]VolumeLockInfo) error
+	GetRWLock(bucket, object string) (RWLocker, error)
```

Comment:
what we had decided was global-NS-lock and object-layer NS-lock will be different entities with no connection.

i.e at the S3/admin layer it will use globalNSMutex and it will never need to call GetRWLock()